### PR TITLE
ref(js): Convert Indicators to a function component

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/components/indicators.tsx [hunk 1]
```diff
@@ -15,68 +13,24 @@ const Toasts = styled('div')`
...
-  }
-
-  unlistener = IndicatorStore.listen((items: Indicator[]) => {
-    this.setState({items});
-  }, undefined);
-
-  render() {
-    // #NEW-SETTINGS - remove ThemeProvider here once new settings is merged
-    // `alerts.html` django view includes this container and doesn't have a theme provider
-    // not even sure it is used in django views but this is just an easier temp solution
```
Comment (evanpurkhiser) [inline static/app/components/indicators.tsx:72]: hmm, I should make sure this wasn't.. important

File: static/app/components/indicators.tsx [hunk 2]
```diff
@@ -15,68 +13,24 @@ const Toasts = styled('div')`
...
-    return (
-      <ThemeProvider theme={lightTheme}>
-        <Indicators {...this.props} items={this.state.items} />
-      </ThemeProvider>
-    );
-  }
+type Props = React.ComponentProps<typeof Toasts>;
```
Comment (priscilawebdev) [inline static/app/components/indicators.tsx:16]: do we want to provide props like `theme` and `as` too? 

```
type Props = {
    theme?: Theme | undefined;
    as?: ElementType<any> | undefined;
} & ClassAttributes<HTMLDivElement> & HTMLAttributes<HTMLDivElement> & {
    ...;
```

shouldn't we only provide className?

Comment (evanpurkhiser) [reply static/app/components/indicators.tsx:16]: Yeah that's fine

File: static/app/components/indicators.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/components/indicators.tsx:72]: Ok. pretty sure it's fine

**Review by priscilawebdev — APPROVED** [review]
left 1 suggestion but it looks good to me

**Comment by evanpurkhiser** [comment]
will push that change up. need to fix tests still

**Pull request merged**